logstash-core/lib/logstash/instrument/snapshot.rb

```diff
@@ -14,8 +14,8 @@ def initialize
     end
 
     def push(*args)
-      type, key, _ = args
-      metric = @metrics.fetch_or_store(key, concrete_class(type, key))
+      namespace, key, type, _ = args
+      metric = @metrics.fetch_or_store([namespace, key].join('-'), concrete_class(type, key))
```

Comment:
good job here with the ```join('-')``` I wonder if we should really do it here, or we can have it pre computed? I imagine this join being something expensive at the end of the day. what do you think?